src/lib/openjp2/pi.c

```diff
@@ -1237,7 +1238,13 @@ opj_pi_iterator_t *opj_pi_create_decode(opj_image_t *p_image,
 	l_current_pi = l_pi;
 
 	/* memory allocation for include */
-	l_current_pi->include = (OPJ_INT16*) opj_calloc((l_tcp->numlayers +1) * l_step_l, sizeof(OPJ_INT16));
+	/* prevent an integer overflow issue */
+	l_current_pi->include = 00;
+	if (l_step_l && l_tcp->numlayers < UINT_MAX / l_step_l - 1)
```

Comment:
Hi detonin and stweil, thanks for your reviewing. So which form is more preferable?
I agree that ```1<<15``` is too small. I'll change the code of this pull request once we come to a conclusion.